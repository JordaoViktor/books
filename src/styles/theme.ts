export default {
  colors: {
    primary: "#FFFF",
    secondary: "#333333",
    tertiary: "#AB2680",
    background: "#E5E5E5",
    dark: "#000",

    lightenText: "#FFFF",
    darkText: "#333333",

    darkOpacity200: "rgba(0,0,0,0.20)",
    darkOpacity300: "rgba(0,0,0,0.32)",
    darkOpacity400: "rgba(0,0,0,0.4)",

    transparent: "transparent"
  },
  fonts: {
    light: "Heebo_300Light",
    normal: "Heebo_400Regular",
    bold: "Heebo_500Medium",
  },
}