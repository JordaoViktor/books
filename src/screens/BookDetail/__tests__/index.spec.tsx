import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import theme from '@styles/theme'
import { BookDetail } from '../index';
import createSagaMiddleware from 'redux-saga';
import configureStore from 'redux-mock-store'
import mockAxios from 'jest-mock-axios';
import { Provider } from 'react-redux';

import { useRoute } from "@react-navigation/core";
jest.mock("@react-navigation/core");
interface IProviderProps {
  children?: React.ReactNode;
  store?: any;
}

const ReduxProvider = ({ children }: IProviderProps) => {
  const initialState = {
    data: {
      data: null
    },
    error: null,
    statusCode: null,
    isLoading: false
  }
  const sagaMiddleware = createSagaMiddleware();

  const mockStore = configureStore([sagaMiddleware])(initialState)

  return (
    <Provider store={mockStore} >
      <ThemeProvider theme={theme} >
        {children}
      </ThemeProvider>
    </Provider>
  )
};

describe('BookDetail Screen', () => {
  test('matches BookDetail screen snapshot', async () => {
    afterEach(() => {
      // cleaning up the mess left behind the previous test
      mockAxios.reset();
    });
    afterAll(() => { jest.useFakeTimers() })
    afterEach(cleanup)
    const navigation = { navigate: jest.fn() };

    const tree = await renderer
      .create(
        <ReduxProvider >
          <BookDetail />
        </ReduxProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  });

})