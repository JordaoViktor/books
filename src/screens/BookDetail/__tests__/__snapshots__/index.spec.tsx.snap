// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetail Screen matches BookDetail screen snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 36,
          "marginTop": 30,
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 35,
          "marginLeft": 16,
          "opacity": 1,
          "width": 35,
        }
      }
    >
      <View
        accessibilityLabel="Previous page"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFF",
            "borderColor": "rgba(0,0,0,0.32)",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "paddingBottom": 9,
            "paddingLeft": 9,
            "paddingRight": 9,
            "paddingTop": 9,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 0,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 0,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "elevation": 5,
              "height": 351,
              "justifyContent": "center",
              "marginTop": 12,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 8,
                "width": 3,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 10.84,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Array [
              Object {
                "height": "100%",
                "width": 240,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#333333",
                "flexWrap": "wrap",
                "fontFamily": "Heebo_500Medium",
                "fontSize": 69,
                "marginTop": 24,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#AB2680",
                "fontFamily": "Heebo_400Regular",
                "fontSize": 35,
                "lineHeight": 20,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 35,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginBottom": 15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            INFORMAÇÕES
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Páginas
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Editora
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Publicação
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Idioma
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Título Original
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            ISBN-10
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            ISBN-13
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            Categoria
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(0,0,0,0.32)",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 30,
              "marginTop": 18,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "marginBottom": 15,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo_500Medium",
                  "fontSize": 24,
                  "lineHeight": 20,
                },
              ]
            }
          >
            RESENHA DA EDITORA
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "rgba(0,0,0,0.32)",
                "fontFamily": "Heebo_500Medium",
                "fontSize": 24,
                "lineHeight": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginRight": 5,
                },
              ]
            }
          >
            SvgMock
          </View>
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
